### PR TITLE
Zach - Toggle Role on Users Page Login Bug

### LINKED ISSUE #3: Login Bug
STR:

 
(1) Try toggling the admin status of a non-admin user.

* Observed: The user does not have admin until they log out and then log back in
* Desired: The user gets admin access immediately.

Related Bug:

Steps to reproduce:

(2) Try toggling an admin user that is not listed in the `ADMIN_USERS` variable to take away their admin access:

* Observed: The user does not lose admin access immediately; instead, they retain it until they log out and log back in again.
* Desired: The user loses admin access immediately.

- [x] Reproduce the login bug
- [x] Debug
- [x] Included an interceptor that runs on all http requests that refreshes roles following a post [here](https://twin.sh/articles/24/spring-security-reload-authorities-on-every-request)
- [x] Included tests following a post [here](https://medium.com/@nooneypradeep/spring-boot-interceptor-with-unit-testing-825cda41f618) 
- [x] Full test coverage + mutation testing
- [x] Functions on localHost + qa for the following cases: 
(1) A user with admin email in OAuth configuration gets admin toggled -> admin role temporarily disabled but restored on refresh
(2) A user not like in (1) does not have admin role but gets promoted -> refresh gives them new privileges
(3) A user not like in (1) has admin role and gets demoted -> refresh removes privileges

### PR DESCRIPTION
**Pull request by zachmiller-ucsb**
Closes #3 , 
Also removes method findByShift and changes findByDriver to findByDriverID in Shift Repository so that Spring can autogenerate those methods. This patches main so that actions will run properly for this PR. 

Watch the first 50 seconds of the following video for a demo on qa of the 3 acceptance criteria on #3 

https://github.com/ucsb-cs156-s23/proj-gauchoride-s23-5pm-4/assets/79069964/56189c58-2e70-43bc-acbc-70c34af93403

### CODE REVIEW

**Review by mike-k999 — APPROVED** [review]
lgtm

File: src/main/java/edu/ucsb/cs156/gauchoride/repositories/ShiftRepository.java [hunk 1]
```diff
@@ -11,6 +11,5 @@
 @Repository
 public interface ShiftRepository extends CrudRepository<Shift, Long> {
   Optional<Shift> findByDay(String day);
-  Optional<Shift> findByDriver(Long driverID);
-  Optional<Shift> findByShift(LocalTime shiftStart, LocalTime shiftEnd);
+  Optional<Shift> findByDriverID(Long driverID);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/repositories/ShiftRepository.java:14]: Is this change supposed to be in this PR?

It seems unrelated...

**Review by pconrad — CHANGES_REQUESTED** [review]
Could you take a look at whether the ShiftRepository.java change is supposed to be in this PR?

**Comment by zachmiller-ucsb** [comment]
This was also patching an issue with main causing gh actions to fail on this commit @pconrad . I can make that clear in the PR message

**Review by SophiaWeiler — APPROVED** [review]
lgtm

**Review by pconrad — APPROVED** [review]
lgtm!

**Pull request merged**